float32 bicep
float32 tricep